homeassistant/components/mqtt/client.py

```diff
@@ -543,21 +649,24 @@ def _client_unsubscribe(topic: str) -> int:
             _raise_on_error(result)
             return mid
 
+        if self._is_active_subscription(topic):
+            # Other subscriptions on topic remaining - don't unsubscribe.
+            return
+        async with self._pending_subscriptions_lock:
+            if topic in self._max_qos:
+                del self._max_qos[topic]
```

Comment:
We cleanup the `_max_qos[topic]` here as we can do it thread safe under the self._pending_subscriptions_lock.